### PR TITLE
Move initialization functions into `esp-hal-common` package's `soc` module

### PR DESCRIPTION
**Pull request by jessebraham**
Creeping ever closer to making #446 a reality... 😅

### CODE REVIEW

File: esp-hal-common/src/soc/esp32/mod.rs [hunk 1]
```diff
@@ -23,3 +26,57 @@ pub(crate) mod constants {
...
     pub const SOC_DRAM_HIGH: u32 = 0x4000_0000;
 }
+
+/// Function initializes ESP32 specific memories (RTC slow and fast) and
+/// then calls original Reset function
+///
+/// ENTRY point is defined in memory.x
+/// *Note: the pre_init function is called in the original reset handler
+/// after the initializations done in this function*
+#[cfg(feature = "rt")]
```
Comment (bjoernQ) [inline esp-hal-common/src/soc/esp32/mod.rs:36]: there is no such feature in esp-hal-common so we don't get ESP32Reset function - makes the code crash immediately

File: esp-hal-common/src/soc/esp32s2/mod.rs [hunk 2]
```diff
@@ -26,3 +30,57 @@ pub(crate) mod constants {
...
     pub const SOC_DRAM_HIGH: u32 = 0x4000_0000;
 }
+
+/// Function initializes ESP32 specific memories (RTC slow and fast) and
+/// then calls original Reset function
+///
+/// ENTRY point is defined in memory.x
+/// *Note: the pre_init function is called in the original reset handler
+/// after the initializations done in this function*
+#[cfg(feature = "rt")]
```
Comment (bjoernQ) [inline esp-hal-common/src/soc/esp32s2/mod.rs:40]: there is no such feature in esp-hal-common so we don't get ESP32Reset function - makes the code crash immediately

File: esp-hal-common/src/soc/esp32s3/mod.rs [hunk 3]
```diff
@@ -29,3 +33,89 @@ pub(crate) mod constants {
...
+    );
+}
+
+/// Function initializes ESP32S3 specific memories (RTC slow and fast) and
+/// then calls original Reset function
+///
+/// ENTRY point is defined in memory.x
+/// *Note: the pre_init function is called in the original reset handler
+/// after the initializations done in this function*
+#[cfg(feature = "rt")]
```
Comment (bjoernQ) [inline esp-hal-common/src/soc/esp32s3/mod.rs:72]: there is no such feature in esp-hal-common so we don't get ESP32Reset function - makes the code crash immediately

(also in other places)

**Comment by bjoernQ** [comment]
Not sure if we should add an `rt` feature to esp-hal-common or not (probably yes )

**Comment by jessebraham** [comment]
Oops, missed some files... 😁

**Comment by jessebraham** [comment]
I decided to handle those features properly instead of what I had hacked together before, so that was taken care of in #1059. I've rebased on `main` since that PR was merged, so I think we should be okay here now.

**Review by bjoernQ — APPROVED** [review]
LGTM!

**Pull request merged**